### PR TITLE
refactor: remove global_eval.ts

### PR DESCRIPTION
**Pull request by kt3k**
This PR replaced the `globalEval` usages with exported `window` object. I created `//js/window.ts` file to separate `const window` declaration from other globals settings and avoid circular reference.

### CODE REVIEW

File: js/window.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+// If you use the eval function indirectly, it works in the global scope rather than
+// the local scope.
+// See https://stackoverflow.com/questions/14119988/return-this-0-evalthis/14120023#14120023
+// TODO: this should be replaced with globalThis
+// when the proposal goes to stage 4
+// See https://github.com/tc39/proposal-global
+export const window = (0, eval)("this");
```
Comment (ry) [inline js/window.ts]: What does this mean? `(0, eval)` ?

Comment (dhcmrlchtdj) [reply js/window.ts]: it is indirect eval
https://tc39.github.io/ecma262/#sec-performeval

Comment (kt3k) [reply js/window.ts]: Yes, it's indirect call of eval. Basically the same as the previous code. Please see the stackoverflow link above for details.

Comment (ry) [reply js/window.ts]: Please add a comment.. Something like this
```
// (0, eval) is indirect eval https://tc39.github.io/ecma262/#sec-performeval
```

**Comment by kt3k** [comment]
Updated the comments. Fixed the lint errors.

File: js/window.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+// (0, eval) is indirect eval.
+// See the links below for details:
+// - https://stackoverflow.com/a/14120023
+// - https://tc39.github.io/ecma262/#sec-performeval (spec)
+export const window = (0, eval)("this");
+// TODO: The above should be replaced with globalThis
```
Comment (kitsonk) [inline js/window.ts]: `globalThis` is already in the runtime... We should be able to do this now, but when I tried it before, there was some sort of error in the snapshotting, I don't know if the snapshot is being done with the right config or some other issue.

**Comment by kt3k** [comment]
@kitsonk 
I've seen [this slide](https://docs.google.com/presentation/d/1u3kbSc5Iga6lG7xoiilTEe2Z59Snz5xNumZaASwSHCI/edit#slide=id.p) which seems presented in TC39 in last January. It proposes the new name `Global` for the global object, and the slide seems saying that the champion (Jordan Harband) is still open to that change. So I'm afraid it's still not completely fixed, and it's safe to do it after it reaches stage 4.

**Comment by kitsonk** [comment]
@kt3k from the notes: https://tc39.github.io/tc39-notes/2019-01_jan-31.html#globalthis-follow-up it was decided to keep `globalThis` and not consider `Global`.  It isn't going to change at this stage because it is already shipped.  The conclusion is that the process is broken and the secrecy around the name was not good and it is akin to smooshgate.

TypeScript is just about to implement it into master: https://github.com/Microsoft/TypeScript/pull/29332

**Comment by kt3k** [comment]
@kitsonk Thank you for the pointer! OK. Let's replace it soon! 👍

**Pull request merged**